src/transformers/trainer.py

```diff
@@ -2258,6 +2259,10 @@ def _inner_training_loop(
         # FSDP-XLA, SageMaker MP/DP, DataParallel, IPEX
         use_accelerator_prepare = True if model is self.model else False
 
+        if use_accelerator_prepare and self.is_fsdp_enabled:
+            # Remove FSDP wrapping from sub-models.
+            self.model = extract_model_from_parallel(self.model, recursive=True)
+
```

Comment:
You can use `unwarp_model` function in transformers instead. Also, why do we need to set `recursive` to `True` ? Also, please leave a comment above as this specific path is only to make it functional with `auto_batch_size`.

Reply:
Unwrap_model do not give  access to recursive argument. The auto wrap policies wrap sub modules with fsdp and unwrap_model is unable to remove it. Y can try to apply unwrap_model on the toy example from https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html for rank=0 and world_size = 1. Then experiment with the line I have written.

Reply:
I'm talking about this [function](https://github.com/huggingface/transformers/blob/ef976a7e181b78abf2f1ba7ea02e506ea1cb111e/src/transformers/modeling_utils.py#L5464) in transformers. It uses extract_model_from_parallel under the hood so it should be comparable.

Reply:
Euh I see.